lib/base_feed.js

```diff
@@ -33,9 +33,32 @@ class BaseFeed {
     }
 
     _hydrateResponse(hyper, req, res) {
+        const rh = Object.assign({}, req.headers);
+        // we don't store /feed/featured no need for cache-control (?)
+        if (rh.host) {
+            /*
+                Keeping the host header makes the fetchSumary fail with HTTPError 504
```

Comment:
@Pchelolo I'm guessing it is not safe to delete host header, and probably is best to modify this header in the test, but I need an insight here because of my lack of knowledge on restbase.

Reply:
```
{ HTTPError: Hostname/IP does not match certificate's altnames: Host: localhost. is not in the cert's altnames: DNS:*.m.wikibooks.beta.wmflabs.org, DNS:*.m.wikimedia.beta.wmflabs.org, DNS:*.m.w
ikinews.beta.wmflabs.org, DNS:*.m.wikipedia.beta.wmflabs.org, DNS:*.m.wikiquote.beta.wmflabs.org, DNS:*.m.wikisource.beta.wmflabs.org, DNS:*.m.wikiversity.beta.wmflabs.org, DNS:*.m.wikivoyage.beta.wmflabs.org, DNS:*.m.wiktionary.beta.wmflabs.org, DNS:*.wikibooks.beta.wmflabs.org, DNS:*.wikimedia.beta.wmflabs.org, DNS:*.wikinews.beta.wmflabs.org, DNS:*.wikipedia.beta.wmflabs.org, DNS:*.wikiquote.beta.wmflabs.org, DNS:*.wikisource.beta.wmflabs.org, DNS:*.wikiversity.beta.wmflabs.org, DNS:*.wikivoyage.beta.wmflabs.org, DNS:*.wiktionary.beta.wmflabs.org, DNS:*.zero.wikipedia.beta.wmflabs.org, DNS:beta.wmflabs.org, DNS:m.wikidata.beta.wmflabs.org, DNS:upload.beta.wmflabs.org, DNS:wikidata.beta.wmflabs.org
    at request.then (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/preq/index.js:246:19)
    at tryCatcher (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/promise.js:725:18)
    at _drainQueueStep (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/async.js:93:12)
    at _drainQueue (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/async.js:86:9)
    at Async._drainQueues (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/async.js:102:5)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/mateus/workspace/wikimedia/infrastructure/restbase/node_modules/bluebird/js/release/async.js:15:14)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)

name: 'HTTPError',
  message:
   'Hostname/IP does not match certificate\'s altnames: Host: localhost. is not in the cert\'s altnames: DNS:*.m.wikibooks.beta.wmflabs.org, DNS:*.m.wikimedia.beta.wmflabs.org, DNS:*.m.wikinews.beta.wmflabs.org, DNS:*.m.wikipedia.beta.wmflabs.org, DNS:*.m.wikiquote.beta.wmflabs.org, DNS:*.m.wikisource.beta.wmflabs.org, DNS:*.m.wikiversity.beta.wmflabs.org, DNS:*.m.wikivoyage.beta.wmflabs.org, DNS:*.m.wiktionary.beta.wmflabs.org, DNS:*.wikibooks.beta.wmflabs.org, DNS:*.wikimedia.beta.wmflabs.org, DNS:*.wikinews.beta.wmflabs.org, DNS:*.wikipedia.beta.wmflabs.org, DNS:*.wikiquote.beta.wmflabs.org, DNS:*.wikisource.beta.wmflabs.org, DNS:*.wikiversity.beta.wmflabs.org, DNS:*.wikivoyage.beta.wmflabs.org, DNS:*.wiktionary.beta.wmflabs.org, DNS:*.zero.wikipedia.beta.wmflabs.org, DNS:beta.wmflabs.org, DNS:m.wikidata.beta.wmflabs.org, DNS:upload.beta.wmflabs.org, DNS:wikidata.beta.wmflabs.org',
  status: 504,
  headers: { 'content-type': 'application/problem+json' },
  body:
   { type: 'internal_http_error',
     detail:
      'Hostname/IP does not match certificate\'s altnames: Host: localhost. is not in the cert\'s altnames: DNS:*.m.wikibooks.beta.wmflabs.org, DNS:*.m.wikimedia.beta.wmflabs.org, DNS:*.m.wikinews.beta.wmflabs.org, DNS:*.m.wikipedia.beta.wmflabs.org, DNS:*.m.wikiquote.beta.wmflabs.org, DNS:*.m.wikisource.beta.wmflabs.org, DNS:*.m.wikiversity.beta.wmflabs.org, DNS:*.m.wikivoyage.beta.wmflabs.org, DNS:*.m.wiktionary.beta.wmflabs.org, DNS:*.wikibooks.beta.wmflabs.org, DNS:*.wikimedia.beta.wmflabs.org, DNS:*.wikinews.beta.wmflabs.org, DNS:*.wikipedia.beta.wmflabs.org, DNS:*.wikiquote.beta.wmflabs.org, DNS:*.wikisource.beta.wmflabs.org, DNS:*.wikiversity.beta.wmflabs.org, DNS:*.wikivoyage.beta.wmflabs.org, DNS:*.wiktionary.beta.wmflabs.org, DNS:*.zero.wikipedia.beta.wmflabs.org, DNS:beta.wmflabs.org, DNS:m.wikidata.beta.wmflabs.org, DNS:upload.beta.wmflabs.org, DNS:wikidata.beta.wmflabs.org',
     internalStack:
      'Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate\'s altnames: Host: localhost. is not in the cert\'s altnames: DNS:*.m.wikibooks.beta.wmflabs.org, DNS:*.m.wikimedia.beta.wmflabs.org, DNS:*.m.wikinews.beta.wmflabs.org, DNS:*.m.wikipedia.beta.wmflabs.org, DNS:*.m.wikiquote.beta.wmflabs.org, DNS:*.m.wikisource.beta.wmflabs.org, DNS:*.m.wikiversity.beta.wmflabs.org, DNS:*.m.wikivoyage.beta.wmflabs.org, DNS:*.m.wiktionary.beta.wmflabs.org, DNS:*.wikibooks.beta.wmflabs.org, DNS:*.wikimedia.beta.wmflabs.org, DNS:*.wikinews.beta.wmflabs.org, DNS:*.wikipedia.beta.wmflabs.org, DNS:*.wikiquote.beta.wmflabs.org, DNS:*.wikisource.beta.wmflabs.org, DNS:*.wikiversity.beta.wmflabs.org, DNS:*.wikivoyage.beta.wmflabs.org, DNS:*.wiktionary.beta.wmflabs.org, DNS:*.zero.wikipedia.beta.wmflabs.org, DNS:beta.wmflabs.org, DNS:m.wikidata.beta.wmflabs.org, DNS:upload.beta.wmflabs.org, DNS:wikidata.beta.wmflabs.org\n    at Object.checkServerIdentity (tls.js:239:17)\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1065:27)\n    at TLSSocket.emit (events.js:198:13)\n    at TLSSocket._finishInit (_tls_wrap.js:633:8)',
     internalURI:
      'https://en.wikipedia.beta.wmflabs.org/api/rest_v1/page/summary/Moscow_Central_Circle',
     internalQuery: '{}',
     internalErr:
      'Hostname/IP does not match certificate\'s altnames: Host: localhost. is not in the cert\'s altnames: DNS:*.m.wikibooks.beta.wmflabs.org, DNS:*.m.wikimedia.beta.wmflabs.org, DNS:*.m.wikinews.beta.wmflabs.org, DNS:*.m.wikipedia.beta.wmflabs.org, DNS:*.m.wikiquote.beta.wmflabs.org, DNS:*.m.wikisource.beta.wmflabs.org, DNS:*.m.wikiversity.beta.wmflabs.org, DNS:*.m.wikivoyage.beta.wmflabs.org, DNS:*.m.wiktionary.beta.wmflabs.org, DNS:*.wikibooks.beta.wmflabs.org, DNS:*.wikimedia.beta.wmflabs.org, DNS:*.wikinews.beta.wmflabs.org, DNS:*.wikipedia.beta.wmflabs.org, DNS:*.wikiquote.beta.wmflabs.org, DNS:*.wikisource.beta.wmflabs.org, DNS:*.wikiversity.beta.wmflabs.org, DNS:*.wikivoyage.beta.wmflabs.org, DNS:*.wiktionary.beta.wmflabs.org, DNS:*.zero.wikipedia.beta.wmflabs.org, DNS:beta.wmflabs.org, DNS:m.wikidata.beta.wmflabs.org, DNS:upload.beta.wmflabs.org, DNS:wikidata.beta.wmflabs.org',
     internalMethod: 'get' } }
```

Reply:
I mean, do you ever set the host header explicitly? you shouldn't

Reply:
The only place where we explicitly set host header is under mw_req config parameter: https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/services/wikifeeds/+/master/config.prod.yaml#89

Reply:
Yeah, so that's my point - this req header you are deleting here shouldn't affect anything in theory, shouldn't be set anywhere. So I don't think deleting it should matter.